src/app/query/bitcoin/fees/fee-estimates.hooks.ts

```diff
@@ -1,9 +1,32 @@
 import { logger } from '@shared/logger';
 
-import { useGetBitcoinFeeEstimatesQuery } from './fee-estimates.query';
+import {
+  useGetBitcoinEarnApiFeeEstimatesQuery,
+  useGetBitcoinMempoolApiFeeEstimatesQuery,
+} from './fee-estimates.query';
 
 export function useBitcoinFeeRate() {
-  return useGetBitcoinFeeEstimatesQuery({
+  return useGetBitcoinMempoolApiFeeEstimatesQuery({
     onError: err => logger.error('Error getting bitcoin fee estimates', { err }),
   });
 }
+
+export function useAverageBitcoinFeeRate() {
+  const { data: mempoolApiFeeRates } = useGetBitcoinMempoolApiFeeEstimatesQuery({
+    onError: err => logger.error('Error getting mempool api bitcoin fee estimates', { err }),
+  });
+  const { data: earnApiFeeRates } = useGetBitcoinEarnApiFeeEstimatesQuery({
+    onError: err => logger.error('Error getting earn api bitcoin fee estimates', { err }),
+  });
+
+  if (!mempoolApiFeeRates && !earnApiFeeRates) {
+    return null;
+  }
+
+  // zero values for cases when one api is down
+  return {
+    fastestFee: ((mempoolApiFeeRates?.fastestFee ?? 0) + (earnApiFeeRates?.fastestFee ?? 0)) / 2,
+    halfHourFee: ((mempoolApiFeeRates?.halfHourFee ?? 0) + (earnApiFeeRates?.halfHourFee ?? 0)) / 2,
+    hourFee: ((mempoolApiFeeRates?.hourFee ?? 0) + (earnApiFeeRates?.hourFee ?? 0)) / 2,
+  };
```

Comment:
We've ran into difficulties in the past when parsing data from query hooks, and in the process hiding the react query props, `isLoading` etc.

Another way to do this is with `select`, where we use the result of the first one and do the averaging there.

Check out the options in the post https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option

Also, let's test what happens if one of these APIs is down. If it is, we should still use the values from the working one.

(Another solution might be to make two requests in the fetcher function with `Promise.allSettled`, and use a single query hook).

Reply:
Also, we have a average helper you can use
https://github.com/hirosystems/wallet/blob/9e21971e70443ae74afc2b4a343a477f9002b28b/src/app/common/calculate-averages.ts#L3-L8

Reply:
fixed! thanks for all tips @kyranjamie